broadinstitute/single_cell_portal_core
# Bulk download selection modal (SCP-3356, SCP-2333)

SCP-3356 SCP-2333
![image](https://user-images.githubusercontent.com/2800795/121283174-b2842e00-c8a8-11eb-864e-cd3bceef314f.png)
![image](https://user-images.githubusercontent.com/2800795/121283188-b87a0f00-c8a8-11eb-8f8a-505594109646.png)

This does not yet handle TDR search results, but the APIs should be relatively straightforward to update once the structure of those results and the ways to get to the file paths become clearer.  To that end, I've taken the bulk download APIs from the search controller and moved them into their own controller

On the frontend, I've removed the DownloadProvider, consistent with our restriction of contexts/providers to only globally-applicable stuff (e.g. user info).  I've also started using react-testing-library.  I agree with Eric that the experience is superior to enzyme, so I think we should use it instead of enzyme going forward.

TO TEST:
  0. sign in and do a search
  1. click 'Download'
  2. Observe that a modal appears
  3. select/deselect studies/files/categories, and observe the total size changes
  4. hit 'next' while at least one file is selected
  5. paste the code that appears into your terminal, and confirm the download works
